src/Http/OpenApi/test/OpenApiGeneratorTests.cs

```diff
@@ -0,0 +1,814 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Metadata;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Primitives;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi.Tests;
+
+public class OpenApiOperationGeneratorTests
+{
+    [Fact]
+    public void OperationNotCreatedIfNoHttpMethods()
+    {
+        var operation = GetOpenApiOperation(() => { }, "/", Array.Empty<string>());
+
+        Assert.Null(operation);
+    }
+
+    [Fact]
+    public void UsesDeclaringTypeAsOperationTags()
+    {
+        var operation = GetOpenApiOperation(TestAction);
+
+        var declaringTypeName = typeof(OpenApiOperationGeneratorTests).Name;
+        var tag = Assert.Single(operation.Tags);
```

Comment:
We usually separate Setup, Act and Assert rather than mix the first to in the middle block e.g.
```suggestion
        var declaringTypeName = typeof(OpenApiOperationGeneratorTests).Name;

        var tag = Assert.Single(operation.Tags);
```

Reply:
Good catch!